bitcoin/src/address/error.rs

```diff
@@ -334,7 +334,11 @@ pub struct LegacyAddressTooLongError {
 
 impl LegacyAddressTooLongError {
     /// Returns the invalid legacy address length.
-    pub fn invalid_legcay_address_length(&self) -> usize { self.length }
+    pub fn invalid_legacy_address_length(&self) -> usize { self.length }
+
+    #[doc(hidden)]
```

Comment:
I'm not generally fond of `doc(hidden)`, even for deprecatedd stuff but given that this one is a super obvious typo, it could be worth decreasing doc noise. Also I think that nobody actually used it because it'd likely be reported (though people use autocompletion, so maybe I'm wrong here).

Reply:
May be worth starting a general discussion about the use of `hidden` with deprecated items. I think for typo fixes it's a good idea and that in general it's a bad one (e.g. if we deprecated something because its functionality was replaced with some subtly different thing, users will want to study the docs for both functions to understand the difference). But not sure where the line is.

Reply:
> I think for typo fixes it's a good idea and that in general it's a bad one

I agree and I think this is where the line is.